### PR TITLE
Thermite tweaks.

### LINKED ISSUE #44944: Thermite Component Code Oddity
Let's play a game, kids:

https://github.com/tgstation/tgstation/blob/92e00cb544c3dc914f771ae638fa2d1aba0f8fc3/code/datums/components/thermite.dm#L54-L69

Can YOU figure out what's wrong with this section of code?

Answer:
```
    if(amount >= 50)
        var/burning_time = max(100, 100-amount)
```
(burning_time is always equal to 100 when it's created, regardless of the value of amount)

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Tweaks thermite in a few ways.

The burn length of thermite is now always 10 seconds, and that's actually the intended behaviour.

The thermite component now gets deleted after burning up.

closes #44944

## Why It's Good For The Game

The original code was just really odd. I think this is an improvement over that.

## Changelog
:cl:
fix: Thermite now gets deleted properly after burning up.
/:cl:

### CODE REVIEW

**Comment by CPTANT** [comment]
Does this chance how much is needed to destroy walls?

**Comment by nemvar** [comment]
No. 5 units for a normal wall. 20 for a reinforced.

**Comment by nemvar** [comment]
I did some research into the history of our thermite and apparently it's supposed to be
```
max(100,300 - thermite)
```
But personally, I think it's kinda stupid that more thermite creates a shorter fire. I want to reward people for using the correct dosage of thermite instead of more is better.

**Comment by JJRcop** [comment]
It would be cool to see thermite need a close to exact amount or else it will start burning through the floor.

**Comment by ATH1909** [comment]
> I did some research into the history of our thermite and apparently it's supposed to be
> 
> ```
> max(100,300 - thermite)
> ```
> 
> But personally, I think it's kinda stupid that more thermite creates a shorter fire. I want to reward people for using the correct dosage of thermite instead of more is better.

But wouldn't more thermite burn through a wall faster/more quickly? Or is that not how that works?

**Comment by CRITAWAKETS** [comment]
If there is more thermite then it should burn faster, since there is more thermite to actually burn.

**Comment by nemvar** [comment]
My original idea of more thermite burning longer with more was flawed because the reason why that happens with normal fires (need for air) doesn't exist for thermite, so reaction time should pretty much be constant. The only difference should be time to destruction, which isn't really a factor for us since our thermite melts instantly. But since lower amounts of burning thermite emits less energy its heat spike drops off quicker, which should represented by the fire wall disappearing faster.

So basically, what I'm saying is that the thermite effect should probably have a static length, which I put at 10 seconds, since 1 unit of thermite creating a 29 second (movement blocking) fire seems pretty idiotic.

File: code/datums/components/thermite.dm [hunk 1]
```diff
@@ -58,15 +62,13 @@
...
-		var/burning_time = max(100, 100-amount)
+	if(amount * burn_coeff >= 50)
 		master = master.Melt()
 		master.burn_tile()
 		if(user)
 			master.add_hiddenprint(user)
-		QDEL_IN(fakefire, burning_time)
-	else
-		QDEL_IN(fakefire, 50)
+	QDEL_IN(fakefire, 100)
```
Comment (ninjanomnom) [inline code/datums/components/thermite.dm]:
```suggestion
	QDEL_IN(fakefire, 10 SECONDS)
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]
